### PR TITLE
Develop v1.3

### PR DESCRIPTION
**Pull request by LoykoLina**

### CODE REVIEW

File: RSS Reader/RSS Reader/Services/Feed service/RSSRFeedService.m [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    
+    return [channel autorelease];
+}
+
+
+#pragma mark - Checking for RSS Feed
+
+- (BOOL)isRSSFeed:(NSData *)data {
+    NSString *contentString = [NSString stringWithUTF8String:data.bytes];
+    return [contentString containsString:kRSSFeedMark] ? YES : NO;
```
Comment (AndreiChernyshou) [inline RSS Reader/RSS Reader/Services/Feed service/RSSRFeedService.m]: Just return [contentString containsString:kRSSFeedMark];

File: RSS Reader/RSS Reader/Services/File Service/RSSRFileService.m [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+- (void)saveChannel:(RSSRChannel *)channel completion:(void (^)(NSError *error))completion {
+    NSError *error = nil;
+    NSDictionary *jsonObject = [channel convertToDictionary];
+    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:jsonObject
+                                                       options:kNilOptions
+                                                         error:&error];
+    if (error) {
+        completion(error);
```
Comment (AndreiChernyshou) [inline RSS Reader/RSS Reader/Services/File Service/RSSRFileService.m]: You will hit a crash if completion is nil

File: RSS Reader/RSS Reader/View Controllers/Settings/Presenter/RSSRSettingsPresenter.m [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    self.feedsSettingsView = view;
+}
+
+- (void)retrieveFeedsFromURL:(NSString *)url {
+    __block typeof(self) weakSelf = self;
+    [self.networkService loadDataFromURL:[NSURL URLWithString:url]
+                         completion:^(NSData *data, NSError *error) {
+        if (error) {
+            [weakSelf showError:error];
+        } else {
```
Comment (AndreiChernyshou) [inline RSS Reader/RSS Reader/View Controllers/Settings/Presenter/RSSRSettingsPresenter.m:97]: You don't need this else statement

File: RSS Reader/RSS Reader/View Controllers/Settings/Presenter/RSSRSettingsPresenter.m [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        }
+    }];
+}
+
+- (void)saveChannel:(RSSRChannel *)channel {
+    __block typeof(self) weakSelf = self;
+    [self.fileService saveChannel:channel completion:^(NSError *error) {
+        if (error) {
+            [weakSelf showError:error];
+        } else {
```
Comment (AndreiChernyshou) [inline RSS Reader/RSS Reader/View Controllers/Settings/Presenter/RSSRSettingsPresenter.m]: the same. Just:
if (error) {
             [weakSelf showError:error];
             return;

File: RSS Reader/RSS Reader/View Controllers/Settings/View/Resized Tableview/RSSRSelfResizedTableView.m [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    [super reloadData];
+    
+    [self invalidateIntrinsicContentSize];
+    
+    [self layoutIfNeeded];
+}
+
+- (CGSize)intrinsicContentSize {
+    CGFloat height = MIN(self.contentSize.height, self.maxHeight);
+    self.scrollEnabled = (height == self.maxHeight) ? YES : NO;
```
Comment (AndreiChernyshou) [inline RSS Reader/RSS Reader/View Controllers/Settings/View/Resized Tableview/RSSRSelfResizedTableView.m]: just like this:
 self.scrollEnabled = (height == self.maxHeight);

**Review by AndreiChernyshou — CHANGES_REQUESTED** [review]
Fix comments, please

File: RSS Reader/RSS Reader/View Controllers/Settings/View/Resized Tableview/RSSRSelfResizedTableView.m (hunk 5 above)
Comment (LoykoLina) [reply RSS Reader/RSS Reader/View Controllers/Settings/View/Resized Tableview/RSSRSelfResizedTableView.m]: Done

File: RSS Reader/RSS Reader/View Controllers/Settings/Presenter/RSSRSettingsPresenter.m (hunk 4 above)
Comment (LoykoLina) [reply RSS Reader/RSS Reader/View Controllers/Settings/Presenter/RSSRSettingsPresenter.m]: Done

File: RSS Reader/RSS Reader/View Controllers/Settings/Presenter/RSSRSettingsPresenter.m (hunk 3 above)
Comment (LoykoLina) [reply RSS Reader/RSS Reader/View Controllers/Settings/Presenter/RSSRSettingsPresenter.m:97]: Done

File: RSS Reader/RSS Reader/Services/File Service/RSSRFileService.m (hunk 2 above)
Comment (LoykoLina) [reply RSS Reader/RSS Reader/Services/File Service/RSSRFileService.m]: Fixed

File: RSS Reader/RSS Reader/Services/Feed service/RSSRFeedService.m (hunk 1 above)
Comment (LoykoLina) [reply RSS Reader/RSS Reader/Services/Feed service/RSSRFeedService.m]: Done

File: RSS Reader/RSS Reader/View Controllers/Settings/View/RSSRSettingsController.m [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+        [self showNoFeedsLabel];
+    }
+}
+
+
+#pragma mark - UITableViewDataSource
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    if ([tableView isEqual:self.feedsTableView]) {
+        return 1;
```
Comment (AndreiChernyshou) [inline RSS Reader/RSS Reader/View Controllers/Settings/View/RSSRSettingsController.m:171]: tertiary operator is more preferable here. No need to fix. Just FYI

File: RSS Reader/RSS Reader/View Controllers/Topics list/Presenter/RSSRTopicsListPresenter.m [hunk 7]
```diff
@@ -9,40 +9,55 @@
 #import "RSSRTopicsListPresenter.h"
 #import "RSSRTopic.h"
```
Comment (AndreiChernyshou) [inline RSS Reader/RSS Reader/View Controllers/Topics list/Presenter/RSSRTopicsListPresenter.m:10]: Imports should be ordered alphabetically. No need to fix. Just FYI

**Review by AndreiChernyshou — APPROVED** [review]

**Pull request merged**